extensions/blocks/gathering-tweetstorms/editor.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import useGatherTweetstorm from './use-gather-tweetstorm';
+import { withNotices, ToolbarButton, ToolbarGroup } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import './editor.scss';
+import { BlockControls } from '@wordpress/editor';
+
+/**
+ * Intercepts the registration of the Core Twitter embed block, and adds functionality
+ * to check whether the tweet being embedded is the start of a tweetstorm. If it is, offer
+ * to import the tweetstorm.
+ *
+ * @param {object} blockSettings - The settings of the block being registered.
+ *
+ * @returns {object} The blockSettings, with our extra functionality inserted.
+ */
+const addTweetstormToTweets = blockSettings => {
+	// Bail if this is not the Twitter embed block, or if the hook has been triggered by a deprecation.
```

Comment:
Since the block relies on a connection to the WordPress.com API, maybe we should avoid displaying it on sites that are not connected to WordPress.com yet, such as sites using Jetpack's development mode? This would avoid messages like this one on sites that cannot be connected to WordPress.com:
![image](https://user-images.githubusercontent.com/426388/85782590-f5e09d80-b726-11ea-9a16-21671bf4e459.png)
I think you should be able to get it from the initial state?

`get( getJetpackData(), [ 'jetpack', 'is_active' ], false );`

Reply:
I didn't think blocks loaded when Jetpack isn't connected? 

I've added a clearer message for dev mode in c25778207.

Reply:
Thank you!

> I didn't think blocks loaded when Jetpack isn't connected?

If Development mode is active, we do load blocks. This allows one to test blocks that do not require a connection to WordPress.com from their local environment. That's why we end up having to check for a connection for blocks like the Instagram one:
https://github.com/Automattic/jetpack/blob/5f098a65b1621103d2e9c288009f528425533ec9/extensions/blocks/instagram-gallery/instagram-gallery.php#L25